### ISSUE TITLE
Nan training loss for CRNN

### ISSUE THREAD

**Issue by cpyang123** [ViSonic-NN/muscribe#6]
During the training of the CRNN for the beats predictions, we the training loss gradually decreased from a very large number such as 300 to nan. 
![43ebd5ba891062a2c8c69538d680a99](https://github.com/ViSonic-NN/muscribe/assets/44955110/60e121a6-d443-4574-b30a-35c46cae0902)

The output predictions are also dubious: 
![image](https://github.com/ViSonic-NN/muscribe/assets/44955110/5eb6b31a-54b2-4e96-814d-95daf346b72c)

This might be due to some normalization problem, or something in the loss function. We'll need to experiment with both to find what the issue might be.

**Comment by cpyang123** [comment]
In light of this problem, we'll try some of the following options: 

- Gradient Clipping
- Learning Rate Warm-up
- Increasing Batch-Size
- Investigating other adjustments to the loss functions

**Comment by cpyang123** [comment]
After a tedious amount of investigation, it was found that the NaNs were produced by zero weights and biases of the embedding layer. Added gradient clipping and decreased the weight decay of the optimizer to 0.00001, the problem was mitigated, and the model was able to run for 100 epochs and converge: 
![image](https://github.com/ViSonic-NN/muscribe/assets/44955110/95c04d57-4af7-43e2-bb40-0fca8a79fa8a)

**Comment by cpyang123** [comment]
For the record, multiple other changes were made to the models: 
- increased model rnn layer count from 3 to 4 
- seq_len_sec increased to 150, and decreased lr to 1e-5
- d_model and d_hid were changed back to 256 and 128 respectively
- numerous assertions were added
